### PR TITLE
CF-2072/CF-2073 Helper methods to fetch all data

### PR DESCRIPTION
**Pull request by cronokun**
Helper methods for https://github.com/rightscale/cs_rds/pull/290/files and https://github.com/rightscale/cs_elb/pull/284.
Fetch records in a loop while NextMarker is present.

### CODE REVIEW

File: spec/cloud/aws/base/helpers/cs_helper_methods_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    it "send a request to AWS" do
+      expect(client).to receive(:api).with("TestRequest", {})
+
+      TestModule.fetch_all(client, "TestRequest",
+        item: "Foo/Bar/Car",
+        marker: "Foo/Bar/Marker"
+      )
+    end
+
+    it "sends another request if response if truncated" do
```
Comment (anatoliliotych) [inline spec/cloud/aws/base/helpers/cs_helper_methods_spec.rb]: typo the second if should be is

**Comment by konstantin-dzreev** [comment]
Hi, what is the point of this commit? It does not look generic, but seems to support RDS only.

The idea of the gem was to be as dry as possible. Pagination and similar things (unless they are generic and work for all AWS services) should stay the client responsibility, not the gem.

**Comment by cronokun** [comment]
Hi, @konstantin-dzreev!
Examples in comments may be misleading, but this code is generic. We have duplicated code like this almost in every CS app. So when I had to add new method `fetch_all` (related to https://bookiee.rightscale.com/browse/CF-2072 and https://bookiee.rightscale.com/browse/CF-2073) I decided to move it here so every CS app could use it. If you know better place to put it I'll be glad to hear it.

**Comment by konstantin-dzreev** [comment]
CS apps are RightScale specific, the gem is an open source project and should have nothing RightScale specific. I'd suggest having a shared library that for those apps.

The commit does not look generic to me, because, at least, "Marker" param is RDS specific. For instance, EC2 uses "NextToken", S3 may expect "NextContinuationToken" or "NextPartNumberMarker", etc.
 Plz, remember, the gem also supports amazon MWS (Market web service) and ECS/PA (ECommerceService Product Advertising) APIs which are again slightly different from both RDS and EC2.

From my point of view the generic fetch_all must either support all the services or stay on the client side.

**Comment by cronokun** [comment]
@konstantin-dzreev I see your point. Closing PR.

**Pull request closed without merging**